src/components/ExerciseLayout.js

```diff
@@ -0,0 +1,120 @@
+import React from 'react';
+import { View, ScrollView, Text, StyleSheet } from 'react-native';
+import { RoundTextButton } from '../components/Components';
+import NavigationMenu from '../components/NavigationMenu';
+import { WHITE, GREEN } from '../styles/colors';
+
+
+const styles = StyleSheet.create({
+  hoveringButton: {
+    position: 'absolute',
+    top: 30,
+  },
+  left: {
+    left: 10,
+  },
+  right: {
+    right: 10,
+  },
+  sectionText: {
+    fontSize: 18,
+    fontWeight: "600",
+    color: WHITE,
+  },
+  chapterText: {
+    fontSize: 16,
+    fontWeight: "500",
+    color: WHITE,
+  },
+  menuButton: {
+    margin: 10,
+  }
+});
+
+
+class ExerciseLayout extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      nextExercise: '/test',
+      navigationMenuVisible: false,
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.setState({
+      navigationMenuVisible: false,
+    });
+  }
+
+  toggleNavigationMenu = () => {
+    this.setState({
+      navigationMenuVisible: !this.state.navigationMenuVisible,
+    });
+  }
+
+  render () {
+    const { course, changeExercise, nextExercise, currentExercise } = this.props;
```

Comment:
`changeExercise` gets passed down from `connect` I suppose? Could we do that in this file directly so that one knows where `changeExercises` comes from?